### ISSUE TITLE
"Claim all" button on the rewards page

### ISSUE THREAD

**Issue by tibfox** [dtube/dtube#199]
Suggestion by adetorrent via dtube video: 

A big "claim all" button would be good for people who do not want to distinguish between the posts. It would be quicker, easier and more user friendly.

https://d.tube/#!/v/adetorrent/5hw9ke92pc0

**Comment by techcoderx** [comment]
This will involve adding a new transaction that iterates over all claimable rewards and executes the claim all at once. It will require less bandwidth than making 20 `CLAIM_REWARD` transactions at once.

**Comment by skzap** [comment]
Claiming or not claiming is one of the important decision in the curation game. Not claiming can make people earn more. Also the implementation would be hard and it would become probably the heaviest transaction to execute in avalon.

So overall a bad idea in my opinion

**Comment by brishtiteveja** [comment]
@skzap I disagree with it. This is a psychological issue. "People want to claim and they want those claimable DTC to be added into their account." And this desire is strong.. as even I myself, go on claiming stuff, knowing that not claiming is better for me. So, we should give a choice of auto claim (claim all). People any way, end up scrolling and claiming and feel like it's a bad feature. What we can do is that - have that bulk transaction, but upon clicking the auto claim button, it should ask for "last 5 digit of master password" and a popup saying that "delaying will produce more".. so basically slightly discouraging claiming at once. But if a user still wants to claim all at once, we should provide the feature.. may be also we can charge/gas fee some DTC for heavy transaction.
